lib/pbench/cli/agent/commands/triggers/list.py

```diff
@@ -0,0 +1,80 @@
+"""
+pbench-list-triggers
+
+This script will list all tool triggers from all tool groups, or list tool
+triggers from a specific group.
+
+"""
+import sys
+
+import click
+
+from pbench.agent.utils import setup_logging
+from pbench.cli.agent import CliContext, pass_cli_context
+from pbench.cli.agent.commands.triggers.base import TriggerCommand
+from pbench.cli.agent.options import common_options
+
+
+class TriggerList(TriggerCommand):
+    """List register triggers"""
+
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.logger = setup_logging()
+
+    def execute(self):
+        if not self.pbench_run.exists():
+            # Silently exit if we don't have a pbench run directory.
+            return 1
+
+        if not self.context.group:
+            # list tool triggers in all groups
+            if len(self.groups) == 0:
+                self.logger.error("%s: error fetching list of tool groups", self.name)
+                return 1
+
+            for group in self.groups:
+                trigger = self.gen_tools_group_dir(group) / "__trigger__"
+                if trigger.exists():
+                    with open(trigger) as f:
+                        data = f.read().splitlines()
+                    print("%s: %s" % (group, "  ".join(data)))
```

Comment:
In the `bash` version, the second argument is just a quoted `$(cat __trigger__)` so the records will remain newline separated. You're separating them with whitespace instead, apparently. Why isn't this breaking a test? Or are all the test cases single lines?

Reply:
OK, so if a tool trigger is a single line with `<start>:<end>` then maybe translating the `$(cat ...)` with `.splitlines()` and `.join()` isn't the best transliteration of the bash idiom?

Perhaps simply printing the first/only line and logging if there's more than one would be cleaner/more explicit?